### PR TITLE
Add Yeehaw Keyboard

### PR DESCRIPTION
**Pull request by Tsquash**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/yeehaw/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case YEEHAW:
+        if (record->event.pressed) {
+            SEND_STRING("yeehaw!");
+        } else {
+        }
```
Comment (ridingqwerty) [inline keyboards/yeehaw/keymaps/via/keymap.c:72]:
```suggestion
        }
```

File: keyboards/yeehaw/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        break;
+        
+    case SQUASHKB:
+        if (record->event.pressed) {
+            SEND_STRING("http://squashkb.com");
+        } else {
+        }
```
Comment (ridingqwerty) [inline keyboards/yeehaw/keymaps/via/keymap.c:79]:
```suggestion
        }
```

File: keyboards/yeehaw/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            SEND_STRING("yeehaw!");
+        } else {
+        }
+        break;
+        
+    case SQUASHKB:
+        if (record->event.pressed) {
+            SEND_STRING("http://squashkb.com");
+        } else {
+        }
```
Comment (ridingqwerty) [inline keyboards/yeehaw/keymaps/default/keymap.c:59]:
```suggestion
        }
```

File: keyboards/yeehaw/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+    case YEEHAW:
+        if (record->event.pressed) {
+            SEND_STRING("yeehaw!");
+        } else {
+        }
```
Comment (ridingqwerty) [inline keyboards/yeehaw/keymaps/default/keymap.c:52]:
```suggestion
        }
```

File: keyboards/yeehaw/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            tap_code(KC_VOLD);
+        } else {
+            tap_code(KC_VOLU);
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code(KC_DOWN);
+        } else {
+            tap_code(KC_UP);
+        }
```
Comment (ridingqwerty) [inline keyboards/yeehaw/keymaps/default/keymap.c:42]:
```suggestion
```

Only one encoder, so this is unneeded.  If you were going for a layer thing, we can work that in.

File: keyboards/yeehaw/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            tap_code(KC_VOLD);
+        } else {
+            tap_code(KC_VOLU);
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code(KC_DOWN);
+        } else {
+            tap_code(KC_UP);
+        }
```
Comment (ridingqwerty) [inline keyboards/yeehaw/keymaps/via/keymap.c:62]:
```suggestion
```

Only one encoder so this is unneeded.

File: keyboards/yeehaw/rules.mk [hunk 7]
```diff
@@ -0,0 +1,27 @@
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (ridingqwerty) [inline keyboards/yeehaw/rules.mk:10]:
```suggestion
```

**Review by ridingqwerty — CHANGES_REQUESTED** [review]

**Review by ridingqwerty — APPROVED** [review]

File: keyboards/yeehaw/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,55 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/yeehaw/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/yeehaw/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,75 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/yeehaw/keymaps/via/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/yeehaw/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+#define ____ KC_TRNS
```
Comment (drashna) [inline keyboards/yeehaw/keymaps/via/keymap.c:3]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by ridingqwerty** [comment]
Thanks!

**Pull request merged**